packages/namekit-react/src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * This TruncatedText component is a duplicate of @namehash/nameguard-react
+ * TruncatedText component. This duplicate will, in the near future, be the
+ * only TruncatedText component in the NameKit libraries. The duplicate is
+ * necessary for the moment so that it is included in main branch while
+ * @namehash/nameguard-react still has its TruncatedText: why is that?
+ *
+ * So any imports of @namehash/nameguard-react TruncatedText component
+ * do not break once this change is merged into main! Once this duplicate
+ * is in main, NameHash Labs will remove the TruncatedText component from
+ * @namehash/nameguard-react and update the imports to use this component
+ * instead, the @namehash/namekit-react TruncatedText.
+ *
+ * A ShortCut story was created to track this migration:
+ * https://app.shortcut.com/ps-web3/story/25261/next-steps-on-displayedname-and-truncatedtext
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import { Tooltip } from "./Tooltip";
+import cc from "classcat";
+
+interface TruncatedTextProps {
+  text: string;
+  maxTooltipWidth?: number;
+  maxDisplayWidth?: number;
+  textStylingClasses?: string;
+  tooltipTextStylingClasses?: string;
+  displayTooltipWhenTextOverflows?: boolean;
+}
+
+const DEFAULT_MAX_DISPLAY_WIDTH = 300;
+const DEFAULT_MAX_TOOLTIP_WIDTH = 400;
+
+export const TruncatedText = ({
+  text,
+  textStylingClasses = "",
+  tooltipTextStylingClasses = "",
+  displayTooltipWhenTextOverflows = true,
+  maxTooltipWidth = DEFAULT_MAX_TOOLTIP_WIDTH,
+  maxDisplayWidth = DEFAULT_MAX_DISPLAY_WIDTH,
+}: TruncatedTextProps) => {
+  const invisibleTextWidthTester = useRef<null | HTMLParagraphElement>(null);
+
+  /* 
+    Below state is only true if the text displayed 
+    is longer than text maxDisplayWidth
+  */
+  const [textOverflows, setTextOverflows] = useState<boolean>(false);
+
+  useEffect(() => {
```

Comment:
I'm pretty sure we could modify this `useEffect` so that it only runs when necessary, and uses lists the dependencies. Something like this?

```ts
useEffect(() => {
    if (displayTooltipWhenTextOverflows && invisibleTextWidthTester.current) {
      const textOverflows =
        Math.ceil(invisibleTextWidthTester.current.getBoundingClientRect().width) > maxDisplayWidth;
      setTextOverflows(textOverflows);
    }
  }, [displayTooltipWhenTextOverflows, maxDisplayWidth, text]);
```

Reply:
@notrab Thanks for suggestion :+1:

@FrancoAguzzi What do you think? Please proceed with updating this as you think is best.

Reply:
@notrab I tested the approach suggest with both normal and Slow 3G (which results in different DOM / JS bundle loading times) and it works just fine as the other approach.  

I suggest the mentioned props are added to `useEffect`'s deps array without the removal of `invisibleTextWidthTester` present in `useEffect`'s dependency array:

All these props affect the component mounting: if text, maxDisplayWidth, displayTooltipWhenTextOverflows or invisibleTextWidthTester are reset / remounted at any time, these affect the DOM element of our `TruncatedText`

`invisibleTextWidthTester` is yet important because depending on the custom classes added to the `TruncatedText`, this element **will or not** reach the `maxDisplayWidth`, needing the maths discussed, then.

Your comment was yet constructive for another piece of this Ui component: we were not styling `invisibleTextWidthTester` DOM element with user preferences for `textStylingClasses`! 

_Quick example on why this is important to be done and code was updated:_

> A user defines `textStylingClasses` as `text-[450px]` and `maxDisplayWidth` as 50px for `text` "easy".
> At first, for human eyes, we shall think that "easy" is pretty small word and `50px` are enough for displaying the full text. At a second level thinking layer, the machine might calculate that 450px fontSize is pretty big and 50px are enough for "ea", only! If `displayTooltipWhenTextOverflows` is true, this `TruncatedText` should, then, display a tooltip with the full "easy" text.

With the update, `invisibleTextWidthTester` is now styled as the real `text` will be so we have real security that the tooltip will be displayed correctly if `displayTooltipWhenTextOverflows` is true.

Thank you for your suggestion, @notrab, it was merged with current approach and provided an error case antecipated handling ✅